### ISSUE TITLE
Bug: MD001 pattern fails to detect header

### ISSUE THREAD

**Issue by ndarville** [markdownlint/markdownlint#73]
It seems the pattern fails to detect a hash header immediately following a dash header:

```
59 Miscellaneous
60 -------------
61 ### Works in Progress ###
62
63 #### Majority Chart ####
64 [![Symmetric Stack Chart thumbnail][majority-chart-img]][majority-chart-url]
```

```
README.md:63: MD001 Header levels should only increment by one level at a time
```

I assume it’s because the pattern doesn’t detect the h3, because there’s no empty line between it and the h2.
